src/objfun_darts.py

```diff
@@ -0,0 +1,278 @@
+        self.set_axis()
+
+    def scatter_points(self, dartboard, x):
+        self.plot_flat_dartboard(dartboard)
+        self.ax.scatter(x[0], x[1], -1 * dartboard.evaluate(x), cmap='coolwarm')
+        self.set_axis()
+
+    def plot_points(self, dartboard, x, y):
+        self.plot_flat_dartboard(dartboard)
+        z = []
+        for array in y:
+            z.append(-1*array[0])
+
+        for i in range(1, len(x[0])):
+            self.ax.plot([x[0][i - 1], x[0][i]], [x[1][i - 1], x[1][i]], [z[i - 1], z[i]],
+                         c=[180 / 255, 180 / 255, 180 / 255], linewidth=1)
+        self.ax.scatter(x[0], x[1], z, cmap='coolwarm')
+        self.set_axis()
+
+    def show_numbers(self, dartboard):
+        number_angles = np.arange(0, 2 * math.pi, dartboard.sector_angle)
+        numbers_distance = 0 * number_angles + dartboard.ring_params[-1][0] * 1.2
+        x = numbers_distance * np.cos(number_angles)
+        y = numbers_distance * np.sin(number_angles)
+        for i in range(len(number_angles)):
+            self.ax.text(x[i], y[i], 0, dartboard.sectors[i], horizontalalignment='center', fontsize=12,
+                         color=[180 / 255, 180 / 255, 180 / 255])
+
+    def set_axis(self):
+        self.ax.set_xlabel('Osa: x [mm]')
+        self.ax.set_ylabel('Osa: y [mm]')
+        self.ax.set_zlabel('Osa: Score')
+        self.ax.set_xlim3d(-250, 250)
+        self.ax.set_ylim3d(-250, 250)
+        self.ax.set_zlim3d(0, 60)
+
+        self.ax.view_init(elev=85, azim=-90)
+
+    def circuit_points_meshgrid(self, step_angle, step_dist, radius):
+        f = np.arange(0, 2 * math.pi, step_angle)
+        f = np.append(f, 2 * math.pi - self.epsilon)
+        [alpha, distance] = np.meshgrid(f, np.arange(0, radius, step_dist))
+        # alpha = alpha.flatten()
+        # distance = distance.flatten()
+
+        x = np.multiply(distance, np.cos(alpha))
+        y = np.multiply(distance, np.sin(alpha))
+        return [x, y]
+
+    def draw_ellipse(self, x0, y0, xR, yR):
+        t = np.arange(-1 * math.pi, math.pi, 0.01)
+        x = x0 + xR * np.cos(t)
+        y = y0 + yR * np.sin(t)
+        self.ax.plot(x, y, 0, c='black', linewidth=0.5)
+
+    def plot_flat_dartboard(self, dartboard):
+
+        for i in range(0, len(dartboard.ring_params)):
+            self.draw_ellipse(0, 0, dartboard.ring_params[i][0], dartboard.ring_params[i][0])
+
+        lines_angle = np.arange(dartboard.dartboard_angle, 2 * math.pi, dartboard.sector_angle)
+        lines_start = 0 * lines_angle + dartboard.ring_params[1][0]
+        lines_end = 0 * lines_angle + dartboard.ring_params[-1][0] * 1.3
+
+        start_x, start_y = pol_to_car(lines_angle, lines_start)
+        end_x, end_y = pol_to_car(lines_angle, lines_end)
+
+        for i in range(len(start_x)):
+            self.ax.plot([start_x[i], end_x[i]], [start_y[i], end_y[i]], c='black', linewidth=0.5)
+        self.show_numbers(dartboard)
```

Comment:
It is always recommended to leave an empty line at the end of any file.

(But maybe only the GitHub environment does not show it... please double check).

Reply:
I was probably a little clumsy while finishing the work.